pkg/build/generator/generator.go

```diff
@@ -515,8 +513,8 @@ func generateBuildFromBuild(build *buildapi.Build) *buildapi.Build {
 	return &buildapi.Build{
 		Spec: buildCopy.Spec,
 		ObjectMeta: kapi.ObjectMeta{
-			Name:   getNextBuildNameFromBuild(buildCopy),
-			Labels: buildCopy.ObjectMeta.Labels,
+			GenerateName: buildCopy.Name + "-",
```

Comment:
hmm, it would replace the last part ("-" separated) if it was made up of numbers.